mcodingbot/plugins/highlights.py

```diff
@@ -17,6 +18,24 @@
 MAX_HIGHLIGHT_LENGTH = 32
 
 
+async def has_permission(guild_id: int, user_id: int, channel_id: int) -> bool:
+    member = plugin.app.cache.get_member(guild_id, user_id)
+    if not member:
+        return False
+
+    channel = plugin.app.cache.get_guild_channel(channel_id)
+    if not channel:
+        # must be a thread
+        thread = plugin.app.rest.fetch_channel(channel_id)
+        assert isinstance(
+            thread, (hikari.GuildThreadChannel, hikari.GuildNewsThread)
```

Comment:
can you use `hikari.GuildThreadChannel` here? This `assert` scares me.

Reply:
The only channel type that hikari doesn't cache entirely are threads - if this assert fails then it needs to error, because something is completely broken with the cache.

Reply:
I don't like that the assert relies on the implementation details of the cache to error/not error. Its not an error if some other type is retrieved from the cache anyway thats not one of those two.